scripts/test-ssl-death-alert.py

```diff
@@ -0,0 +1,122 @@
+
+
+def help_msg():
+    print("Usage: <script-name> [-h hostname] [-p port] [-n number_of_alerts]")
+    print(" -h hostname          name of the host to run the test against")
+    print("                      localhost by default")
+    print(" -p port              port number to use for connection,")
+    print("                      4433 by default")
+    print(" -n number_of_alerts  how many alerts client sends to server,")
+    print("                      4 by default")
+    print(" --help               this message")
+
+
+def main():
+    hostname = "localhost"
+    port = 4433
+    number_of_alerts = 4
+
+    argv = sys.argv[1:]
+    opts, args = getopt.getopt(argv, "h:p:n:", ["help"])
+    for opt, arg in opts:
+        if opt == '-h':
+            hostname = arg
+        elif opt == '-p':
+            port = int(arg)
+        elif opt == '-n':
+            number_of_alerts = int(arg)
+        elif opt == '--help':
+            help_msg()
+            sys.exit(0)
+        else:
+            raise ValueError("Unknown option: {0}".format(opt))
+
+    conversations = {}
+
+    conversation = Connect(hostname, port, version=(3, 3))
+    node = conversation
+    ciphers = [CipherSuite.TLS_RSA_WITH_AES_128_CBC_SHA]
+    node = node.add_child(ClientHelloGenerator(ciphers))
+    for _ in range(number_of_alerts):  # sending alerts during handshake
+        node = node.add_child(AlertGenerator(  # alert description: 46, 41, 43
+            AlertLevel.warning, AlertDescription.unsupported_certificate))
+    node = node.add_child(ExpectServerHello())
+    node = node.add_child(ExpectCertificate())
+    node = node.add_child(ExpectServerHelloDone())
+    node = node.add_child(ClientKeyExchangeGenerator())
+    node = node.add_child(ChangeCipherSpecGenerator())
+    node = node.add_child(FinishedGenerator())
+    node = node.add_child(ExpectChangeCipherSpec())
+    node = node.add_child(ExpectFinished())
+    node = node.add_child(AlertGenerator(
+        AlertLevel.warning, AlertDescription.close_notify))
+    node = node.add_child(ExpectAlert(AlertLevel.warning,
+                                      AlertDescription.close_notify))
+    node.next_sibling = ExpectClose()
+    conversations["SSL Death Alert without getting alert"] = conversation
+
+    conversation = Connect(hostname, port, version=(3, 3))
+    node = conversation
+    ciphers = [CipherSuite.TLS_RSA_WITH_AES_128_CBC_SHA]
+    node = node.add_child(ClientHelloGenerator(ciphers))
+    for _ in range(number_of_alerts+1):
+        node = node.add_child(AlertGenerator(
+            AlertLevel.warning, AlertDescription.unsupported_certificate))
+    node = node.add_child(ExpectServerHello())
+    node = node.add_child(ExpectCertificate())
+    node = node.add_child(ExpectServerHelloDone())
+    #for _ in range(number_of_alerts+1):
+    node = node.add_child(ExpectAlert(AlertLevel.fatal))
+    node.next_sibling = ExpectClose()
```

Comment:
`next_sibling` is used to provide alternatives (what we are building here is a directed graph/tree)

in other words, this test case will expect Alert or connection close from the other side, since we are inducing error, that is not correct, connection close should always follow an Alert message, like this:
```
node = node.add_child(ExpectAlert(AlertLevel.fatal))
node = node.add_child(ExpectClose())
```

for close_notify we allow either Alert response or immediate close because not all implementations allow for orderly close of the connection, but it still should expect close after the alert:
```
    node = node.add_child(ExpectAlert(AlertLevel.warning,
                                      AlertDescription.close_notify))
    node.next_sibling = ExpectClose()
    node = node.add_child(ExpectClose())
```